numba/datamodel/models.py

```diff
@@ -491,6 +490,17 @@ def from_data(self, builder, value):
         return NotImplemented
 
 
+@register_default(types.NestedArray)
+class NestedArrayModel(ArrayModel):
+    def __init__(self, dmm, fe_type):
+        self._be_type = dmm.lookup(fe_type.dtype).get_data_type()
+        super(NestedArrayModel, self).__init__(dmm, fe_type)
+
+    def get_data_type(self):
```

Comment:
I'm not sure there is anything wrong with `sentry_record_type` - as I understand it, it is really the case that the data type of a `NestedArray` is the type of the elements. This is consistent with the difference between the way arguments are passed to functions as well - a parameter that is a NestedArray receives a pointer to the data, whereas a parameter that is an Array receives the array struct (or a series of "unpacked" struct members, in the case of CUDA).